packages/firestore/src/local/indexeddb_query_cache.ts

```diff
@@ -87,12 +87,16 @@ export class IndexedDbQueryCache implements QueryCache {
     });
   }
 
-  setLastRemoteSnapshotVersion(
+  setTargetsMetadata(
     transaction: PersistenceTransaction,
-    snapshotVersion: SnapshotVersion
+    highestListenSequenceNumber: number,
+    lastRemoteSnapshotVersion?: SnapshotVersion
```

Comment:
Why is this optional? Do we ever not supply it?

Reply:
Local Store doesn't always call setLastRemoteSnapshotVersion. Do you still want to update the listen sequence number?

See https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/local/local_store.ts#L524

Reply:
I see. I do still want to update the sequence number. I guess optional is ok, although it looks like in that case we could also pass in the highest snapshot version and have it be a no-op. I don't feel too strongly either way though.

Reply:
We can refine the API when we start porting to Web, but I believe the highest snapshot version that you know of could be SnapshotVersion.MIN (unless you pass the version that you just got from getLastRemoteSnapshotVersion).

Reply:
Yeah, I meant the one you just read. But yeah, we can update later if need be.